samza-core/src/main/java/org/apache/samza/util/StreamUtil.java

```diff
@@ -0,0 +1,87 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.util;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.MapConfig;
+import org.apache.samza.config.StreamConfig;
+import org.apache.samza.config.StreamConfig$;
+import org.apache.samza.system.StreamSpec;
+import org.apache.samza.system.SystemStream;
+
+public class StreamUtil {
+  /**
+   * Returns a SystemStream object based on the system stream name given. For
+   * example, kafka.topic would return new SystemStream("kafka", "topic").
+   */
+  public static SystemStream getSystemStreamFromNames(String systemStreamNames) {
+    int idx = systemStreamNames.indexOf('.');
+    if (idx < 0) {
+      throw new IllegalArgumentException("No '.' in stream name '" + systemStreamNames +
+          "'. Stream names should be in the form 'system.stream'");
+    }
+    return new SystemStream(
+        systemStreamNames.substring(0, idx),
+        systemStreamNames.substring(idx + 1, systemStreamNames.length()));
+  }
+
+  /**
+   * Returns a SystemStream object based on the system stream name given. For
+   * example, kafka.topic would return new SystemStream("kafka", "topic").
+   */
+  public static String getNameFromSystemStream(SystemStream systemStream) {
+    return systemStream.getSystem() + "." + systemStream.getStream();
+  }
+
+  public static Set<StreamSpec> getStreamSpecs(Set<String> streamIds, StreamConfig streamConfig) {
+    return streamIds.stream().map(streamId -> getStreamSpec(streamId, streamConfig)).collect(Collectors.toSet());
+  }
+
+  public static StreamSpec getStreamSpec(String streamId, StreamConfig streamConfig) {
+    String physicalName = streamConfig.getPhysicalName(streamId);
+    String system = streamConfig.getSystem(streamId);
+    Map<String, String> streamProperties = streamConfig.getStreamProperties(streamId);
+    return new StreamSpec(streamId, physicalName, system, streamProperties);
+  }
+
+  /**
+   * Converts the provided list of (streamId, system, physicalName) triplets to their corresponding
+   * stream.stream-id.* configurations.
+   *
+   * @param streams a list of (streamId, system, physicalName) triplets to get the stream configuration for.
+   * @return the configuration for the provided { @code streams}
+   */
+  public static Config toStreamConfigs(List<ImmutableTriple<String, String, String>> streams) {
```

Comment:
From what I can tell, this is just used in test code. Should we avoid mixing prod utils and test utils?
Also, would it be clearer to make some explicit container for this set of data instead of using a triple, especially because all of the types are String.

Reply:
Yeah, it's currently only used in tests. There's no general pattern / central place for test utils in this code base so I put it here for discoverability. Do you think it's worth extracting to a single method StreamTestUtils? Does anyone else know of a better place for this?

Agree that the String triple is confusing, but I didn't want to introduce a new holder type in the API just for this. StreamSpec could be used as that holder class but that would be undesirable since the intention here is to remove StreamSpec usage from these classes.

Reply:
In general, I do think that test utils should be separate from prod utils. It helps prevent someone from accidentally using it in prod code in the future when it is not intended for prod code. I think it is reasonable to start a new StreamTestUtils for that if there isn't already a "test utils" class for it.
If you do move it into a test utils class, then it might be more reasonable to use StreamSpec as a holder, since it would just be using it in tests (although maybe your own new holder would be better if something changes in the future). Another way to do it would be to make this only work for a single (id, systemName, physicalName), and then have the caller call it multiple times if they have multiple streams.

Reply:
Is it possible to move it to a separate util package in the `samza-core/src/test/` path?
Also, I wouldn't be against having a new holder type. The benefit of a new type is that it will be self documenting. Not a big concern for me though.

Reply:
Will move to a new test util class. I'd prefer to not introduce StreamSpec back into this package, even for tests. 

How about changing this method to:
`void addStreamConfigs(Map<String, String> configs, String id, String system, String physicalName)`

Would prefer that over adding a tests-only holder class.

Reply:
Sounds good to me.